extensions/markdown-language-features/src/tableOfContentsProvider.ts

```diff
@@ -52,11 +51,12 @@ export class TableOfContentsProvider {
 		const toc: TocEntry[] = [];
 		const tokens = await this.engine.parse(document);
 
+		const lines = toTextLines(document.getText());
```

Comment:
It would be nicer if the logic for getting the lines was encapsulated inside `SkinnyTextDocument` instead. That way, for cases where we do have a `vscode.TextDocument` we can just use `vscode.TextDocument.lines` and only for cases where we are reading a file from the disk do we need to use `toTextLines`

Reply:
How do you want me to do this? SkinnyTextDocument was in this extension before and was an interface. I wanted to minimize the size of the change. Could you suggest preferred structure?

Reply:
I made textDocument provide the same line related fields and methods as TextDocument.